pipeline/lib/storage.ml

```diff
@@ -135,3 +135,33 @@ let mark_closed_pull_requests ~open_pulls (db : Postgresql.connection) =
 
 let mark_closed_pull_requests ~open_pulls ~conninfo =
   Db_util.with_db ~conninfo (mark_closed_pull_requests ~open_pulls)
+
+let get_main_branch_metrics ~repository ~worker ~docker_image
+    (db : Postgresql.connection) =
+  let repo_id = Db_util.string @@ Repository.info repository
+  and commit = Db_util.string @@ Repository.commit_hash repository
+  and worker = Db_util.string @@ worker
+  and docker_image = Db_util.string @@ docker_image in
+  let main_commit =
+    Fmt.str
+      {|(SELECT commit FROM benchmarks
+       WHERE pull_number is NULL AND repo_id=%s AND worker=%s AND docker_image=%s
+       ORDER BY -EXTRACT(epoch FROM run_at) LIMIT 1)
+       |}
+      repo_id worker docker_image
+  in
+  let query_tmpl =
+    Fmt.str
+      {|SELECT commit, benchmark_name, test_name, metrics FROM benchmarks
+       WHERE commit = %s AND repo_id=%s AND worker=%s AND docker_image=%s;
+       |}
+  in
+  let query = query_tmpl main_commit repo_id worker docker_image in
+  let result_main = db#exec query in
+  let query = query_tmpl commit repo_id worker docker_image in
+  let result = db#exec query in
+  (result_main#get_all, result#get_all)
```

Comment:
Nitpick: it might be more verbose, but I think "parsing" the sql results at this stage would help the typechecker catch bugs in the long run... The issue with a `string array array` output is that if the query is updated and returns more/less columns, then the typechecker is not going to complain that we broke the users of this function. The "parsing" can be as simple as:
```ocaml
Array.map (function [| commit ; bench_name ; test_name ; metrics |] ->
                      (commit, bench_name, test_name, metrics) ...
```
(but we might also want to `metrics |> Yojson.Safe.from_string |> Current_bench_json.Latest.metrics_of_json []` to ensure a coherent typing)